src/object/assign.ts

```diff
@@ -14,10 +14,10 @@ import { isPlainObject } from 'radashi'
  * // => { a: 1, b: 2, c: 3, p: { a: 4, b: 5 } }
  * ```
  */
-export function assign<X extends Record<string | symbol | number, any>>(
-  initial: X,
-  override: X,
-): X {
+export function assign<
+  X extends Record<string | symbol | number, any>,
+  Y extends Record<string | symbol | number, any>,
+>(initial: X, override: Y): X & Y {
```

Comment:
Unfortunately, `X & Y` is not representative.

```ts
type A = {a: number}
type B = {a: string}

type C = A & B

declare const c: C
c.a // number & string
```
[Link](https://typescript-play.js.org/#code/C4TwDgpgBAglC8UDeBDAXFAdgVwLYCMIAnAXwChRIoAhBZdKAZ2CIEtMBzci8aAYTpwAZDTJkAJhADGAGxRFoUgPaZmUKRj5kpAOhRQA9Aax5CRKCOZtOQA) to above example in playground.

We actually want `c.a` to result in `string` (as string will always override number). If `B["a"]` was possibly undefined like `{a?: string}`, then `c.a` should result in `number | string`.

Reply:
You're right, `A & B` does not result in the desired type.

Perhaps something like the `Merge` TS helper from https://github.com/millsp/ts-toolbelt/blob/master/sources/Object/Merge.ts could work?